spring-integration-core/src/main/java/org/springframework/integration/context/IntegrationObjectSupport.java

```diff
@@ -191,7 +191,7 @@ protected TaskScheduler getTaskScheduler() {
 		return this.channelResolver;
 	}
 
-	protected void setTaskScheduler(TaskScheduler taskScheduler) {
+	public void setTaskScheduler(TaskScheduler taskScheduler) {
```

Comment:
Maybe it would be better to override this method to `public` like it is in the `AbstractCorrelatingMessageHandler`, for example?
It really smells a bit to publish such kind of options in the `IntegrationObjectSupport`...